src/PrizePool.ts

```diff
@@ -50,26 +53,36 @@ export class PrizePool {
     signerOrProvider: Provider | Signer,
     contractMetadataList: ContractMetadata[]
   ) {
-    // Get contract metadata & ethers contracts
-    const prizePoolContract = new Contract(
-      prizePoolMetadata.address,
-      prizePoolMetadata.abi,
-      signerOrProvider
-    )
-
-    // Set data
+    // Set basic data
     this.contractMetadataList = contractMetadataList
     this.signerOrProvider = signerOrProvider
     this.chainId = prizePoolMetadata.chainId
     this.address = prizePoolMetadata.address
 
+    // Get prizePool ethers contract
+    const prizePoolContract = new Contract(
+      this.address,
+      prizePoolMetadata.abi,
+      this.signerOrProvider
+    )
+
+    // Get eip2612PermitAndDeposit metadata & ethers contracts
+    const eip2612PermitAndDeposit = getMetadataAndContract(
```

Comment:
`getMetadataAndContract` throws an error if there is no EIP2612PermitAndDeposit contract in the contract list. The constructor should still successfully run without that contract.

Reply:
Updated through a5bf64a